company_descriptions/make_training_data/split_job_ads_flow.py

```diff
@@ -0,0 +1,108 @@
+"""
+Clean and split job ads into to sentences in order to label to fine-tune a BERT-like model for job advertisements
+
+python ojd_daps_language_models/pipeline/make_training_data/split_job_ads.py run --chunk_size 1000
+
+"""
+from metaflow import FlowSpec, step, Parameter
+import os
+
+
+class CleanJobsFlow(FlowSpec):
+    chunk_size = Parameter(
+        "chunk_size",
+        help="# of job adverts to include per chunk",
+        default=900,
+    )
+    production = Parameter(
+        "production", help="to run in production mode", default=False
+    )
+
+    @step
+    def start(self):
+        """
+        Starts the flow.
+        """
+        self.next(self.get_job_ads)
+
+    @step
+    def get_job_ads(self):
+        """Loads job ads from S3"""
+        from nesta_ds_utils.loading_saving.S3 import download_obj
+
+        print("loading raw job ads")
+
+        self.raw_job_ads = (
+            download_obj(
+                "open-jobs-lake",
+                os.path.join(
+                    "escoe_extension/outputs/data/model_application_data/",
+                    "raw_job_adverts_sample.csv"
+                ),
+                download_as="dataframe",
+            )
+            .set_axis(
+                ["id", "date", "job_title", "job_description"], axis=1, inplace=False
+            )
+            .drop_duplicates("job_description")
+            .job_description.to_list()
+        )
+
+        self.raw_job_ads if self.production else self.raw_job_ads[:10]
+
+        self.next(self.chunk_job_ads)
+
+    @step
+    def chunk_job_ads(self):
+        """Chunk job ad texts"""
+        from toolz import partition_all
+
+        self.job_ad_chunks = list(partition_all(self.chunk_size, self.raw_job_ads))
+
+        self.next(self.clean_job_ads, foreach="job_ad_chunks")
+
+    @step
+    def clean_job_ads(self):
+        """Cleans job ads by:
+        - removing numbers and stop words
+        - Creating new sentences by Camel Case
+        - splitting sentences
+
+        and saves sentence chunks as .csv files
+        """
+        import uuid
+
+        from nesta_ds_utils.loading_saving.S3 import upload_obj
```

Comment:
Just for my understanding - what's the benefit of importing these libraries within a step, rather than at the top of the script?